app/main.py

```diff
@@ -1,44 +1,67 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    await asyncio.gather(*functions)
```

Comment:
The `run_parallel` function is a good utility function. However, you could use `asyncio.gather` directly instead of wrapping it in another function. It would make your code more readable and straightforward.